homeassistant/components/proxmoxve/sensor.py

```diff
@@ -0,0 +1,126 @@
+"""Sensor to read Proxmox VE data."""
+
+from __future__ import annotations
+
+from collections.abc import Callable
+from dataclasses import dataclass
+
+from homeassistant.components.sensor import SensorEntity, SensorEntityDescription
+from homeassistant.const import DATA_GIBIBYTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from . import COORDINATORS, DOMAIN, PROXMOX_CLIENTS, ProxmoxEntity
+
+BYTE_TO_GIBIBYTE = 1.074e9
+
+
+@dataclass
+class ProxmoxVERequiredKeyMixin:
+    """Mixin for required lambda functions."""
+
+    native_lambda: Callable
+
+
+@dataclass
+class ProxmoxVESensorEntityDescription(
+    SensorEntityDescription, ProxmoxVERequiredKeyMixin
+):
+    """Class to describe a proxmoxve sensor."""
+
+
+SENSOR_DESCRIPTIONS = [
+    (
+        ProxmoxVESensorEntityDescription(
+            key="mem_gib",
+            name="mem_gib",
+            native_unit_of_measurement=DATA_GIBIBYTES,
+            native_lambda=lambda data: round(int(data.mem) / BYTE_TO_GIBIBYTE, 2),
```

Comment:
Please remove the conversion.
Use bytes as native unit and set suggest unit with device class.

Reply:
Made the change. Feel free to resolve this if it's what you expected